instrument-starters/opentracing-spring-cloud-feign-starter/src/main/java/io/opentracing/contrib/spring/cloud/feign/TracingFeignBuilder.java

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright 2017-2019 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.contrib.spring.cloud.feign;
+
+import feign.Client;
+import feign.Contract;
+import feign.ExceptionPropagationPolicy;
+import feign.Feign;
+import feign.InvocationHandlerFactory;
+import feign.Logger;
+import feign.QueryMapEncoder;
+import feign.Request;
+import feign.RequestInterceptor;
+import feign.ResponseMapper;
+import feign.Retryer;
+import feign.Target;
+import feign.codec.Decoder;
+import feign.codec.Encoder;
+import feign.codec.ErrorDecoder;
+
+public class TracingFeignBuilder extends Feign.Builder {
```

Comment:
Instead of wrapping this cannot we do something like this https://github.com/spring-cloud/spring-cloud-sleuth/blob/master/spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/feign/SleuthFeignBuilder.java

Reply:
We could but what if somebody has already put a builder in the context? In that case this code is ignored.

Reply:
I simplified implementation to not wrap Feign.Builder